src/app/main-view/sidebar/sidebar.component.ts

```diff
@@ -1,15 +1,130 @@
+import { distinctUntilChanged } from 'rxjs/operators';
+import { FiltersInterface } from '../../interfaces/filters-interface';
+import { Subject } from 'rxjs';
 
 @Component({
-  selector: 'app-sidebar',
-  templateUrl: './sidebar.component.html',
-  styleUrls: ['./sidebar.component.scss']
+    selector: 'app-sidebar',
+    templateUrl: './sidebar.component.html',
+    styleUrls: ['./sidebar.component.scss']
 })
-export class SidebarComponent implements OnInit {
+export class SidebarComponent implements OnInit, OnDestroy {
+    @Input() products: ProductInterface[];
+    @Output() changeFilter: EventEmitter<FiltersInterface> = new EventEmitter();
 
-  constructor() { }
+    public objectKeys = Object.keys;
+    private unsubscribeAll = new Subject();
 
-  ngOnInit(): void {
-  }
+    private maxPrice: number;
+    public stars: number[] = [5, 4, 3, 2, 1];
+    public options: Options;
+
+    public categories: {};
+    public brands: string[];
+
+    public filtersForm: FormGroup;
+
+    constructor(private filtersService: FiltersService) {
+        this.filtersService.handleFilters(this.filtersForm);
+    }
+
+    ngOnInit(): void {
+        this.categories = this.getUniqueCategories();
+        this.brands = this.getUniqueBrands();
+        this.maxPrice = this.getMaxPrice();
+        this.options = this.initializeOptions();
+        this.filtersForm = this.initializeForm();
+        this.onChanges();
+    }
+
+    ngOnDestroy(): void {
+        this.unsubscribeAll.next();
+        this.unsubscribeAll.complete();
+    }
+
+
+    private initializeForm(): FormGroup {
+        return new FormGroup({
+            category: new FormControl(''),
+            brands: new FormArray(this.brands.map(() => new FormControl(false))),
+            rating: new FormArray(this.stars.map(() => new FormControl(false))),
+            price: new FormControl([0, this.maxPrice])
+        });
+    }
+
+
+    private initializeOptions(): Options {
+        return {
+            floor: 0,
+            ceil: this.maxPrice,
+            step: 1
+        };
+    }
+
+    private onChanges(): void {
+        this.filtersForm.valueChanges.pipe(
+            distinctUntilChanged()
+        ).subscribe(val => {
+            const selectedBrands = [];
+            const selectedRating = [];
+            if (val.brands.includes(true)) {
+                val.brands.map((item, index) => {
+                    if (item === true) {
+                        selectedBrands.push(this.brands[index]);
+                    }
+                });
+            }
+            if (val.rating.includes(true)) {
+                val.rating.map((item, index) => {
+                    if (item === true) {
+                        selectedRating.push(this.stars[index]);
+                    }
+                });
+            }
+            this.filtersService.handleFilters({ ...val, brands: selectedBrands, rating: selectedRating });
```

Comment:
You can probably split this: 
```ts 
if (val.brands.includes(true)) {
    val.brands.map((item, index) => {
        if (item === true) {
            selectedBrands.push(this.brands[index]);
        }
   });
}
```
To method: 
```ts 
private getSelectedBrands(brands: []): [] {
  const selectedBrands: [] = [];
  if (brands.includes(true)) {
    brands.map((item, index) => {
        if (item) {
            selectedBrands.push(this.brands[index]);
        }
     });
  }
 return selectedBrands;
}
```
Do this as well as with other `If`, and as final result you'll get such method: 
```ts
private onChanges(): void {
  this.filtersForm.valueChanges.pipe(distinctUntilChanged()).subscribe(
    val => {
      this.filtersService.handleFilters({ ...val, brands: this.getSelectedBrands(val.brands), rating: this.getSelectedRating(val.rating);
      this.changeFilter.emit();
    }
);
```

Reply:
Fixed